ground/gcs/src/plugins/coreplugin/mainwindow.cpp

```diff
@@ -331,31 +317,31 @@ void MainWindow::extensionsInitialized()
             qDebug() << "Looking for default config files in: " + directory.absolutePath();
         bool showDialog=true;
         QString filename;
-        if(!commandLine.isEmpty())
-        {
-            if(QFile::exists(directory.absolutePath()+QDir::separator()+commandLine))
-            {
-                filename=directory.absolutePath()+QDir::separator()+commandLine;
-                emit splashMessages(tr("Loading configuration from command line"));
-                qDebug()<<"Load configuration from command line";
-                settings=new QSettings(filename, XmlConfig::XmlSettingsFormat);
-                showDialog=false;
-            }
-        }
         if(showDialog)
         {
-            importSettings * dialog=new importSettings(this);
+            importSettings * dialog = new importSettings(this);
             dialog->loadFiles(directory.absolutePath());
             dialog->exec();
-            filename=dialog->choosenConfig();
-            settings=new QSettings(filename, XmlConfig::XmlSettingsFormat);
+            filename = dialog->choosenConfig(usePortableSettings);
+            settings = new QSettings(filename, XmlConfig::XmlSettingsFormat);
             delete dialog;
         }
-        qs=settings;
+        qs = settings;
         emit splashMessages(QString(tr("Loading default configuration from %1")).arg(filename));
         qDebug() << "Load default config from resource "<<filename;
     }
     qs->beginGroup("General");
+    if(usePortableSettings) {
+        bool writable;
+        Utils::PathUtils::getInstance()->getLocalSettingsFilePath(writable);
+        if(!writable) {
+            QMessageBox::warning(this, tr("Settings directory no writable"), tr("Reverting to global settings"));
```

Comment:
It'd be nice if this was shared between the two places with similar code somehow.

Also I suggest "Settings directory is not writable"